src/authplugins/ProxyFirstBasic.cpp

```diff
@@ -0,0 +1,88 @@
+// ProxyFirstBasic auth plugin
+//
+// Plugin for use where e2g is behind squid and squid sends user in basic 
+// format
+
+// For all support, instructions and copyright go to:
+// http://e2guardian.org/
+// Released under the GPL v2, with the OpenSSL exception described in the README file.
+
+// INCLUDES
+#ifdef HAVE_CONFIG_H
+#include "e2config.h"
+#endif
+
+#include "../Auth.hpp"
+#include "../OptionContainer.hpp"
+
+#include <syslog.h>
+
+extern bool is_daemonised;
+extern OptionContainer o;
+extern thread_local std::string thread_id;
+
+// DECLARATIONS
+
+// class name is relevant!
+class pf_basic_instance : public AuthPlugin
+{
+    public:
+    pf_basic_instance(ConfigVar &definition)
```

Comment:
Codacy found an issue: [Class 'pf_basic_instance' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238856)